packages/swapper/src/swappers/RelaySwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,242 @@
+      destinationChainId: buyRelayChainId,
+      destinationCurrency: getRelayEvmAssetAddress(buyAsset),
+      tradeType: 'EXACT_INPUT',
+      amount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      recipient: receiveAddress,
+      user: sendAddress,
+      slippageTolerance: slippageToleranceBps,
+    },
+    deps.config,
+  )
+
+  if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
+
+  const quote = maybeQuote.unwrap()
+
+  const swapSteps = quote.data.steps.filter(step => step.id !== 'approve')
+  const hasApprovalStep = quote.data.steps.find(step => step.id === 'approve')
+
+  if (swapSteps.length > 2) {
+    deps.mixPanel?.track(MixPanelEvent.RelayMultiSteps, {
+      swapper: SwapperName.Relay,
+      method: 'get',
+    })
+
+    return Err(
+      makeSwapErrorRight({
+        message: `Relay quote with ${swapSteps.length} swap steps not supported (maximum 2)`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  const slippageTolerancePercentageDecimal = (() => {
+    if (_slippageTolerancePercentageDecimal) return _slippageTolerancePercentageDecimal
+    const destinationSlippageTolerancePercentageDecimal = bnOrZero(
+      quote.data.details?.slippageTolerance?.destination?.percent,
+    )
+
+    if (destinationSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(destinationSlippageTolerancePercentageDecimal).toFixed()
+    }
+
+    const originSlippageTolerancePercentageDecimal = bnOrZero(
+      quote.data.details?.slippageTolerance?.origin?.percent,
+    )
+
+    if (originSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(originSlippageTolerancePercentageDecimal).toFixed()
+    }
+  })()
+
+  const relayToken = quote.data.fees?.relayerService?.currency
+
+  if (!relayToken) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay protocol token not found',
+      }),
+    )
+  }
+
+  if (!isRelayToken(relayToken)) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay protocol token is not properly typed',
+      }),
+    )
+  }
+
+  const protocolAssetId = relayTokenToAssetId(relayToken)
+
+  const protocolAsset = relayTokenToAsset(relayToken, deps.assetsById)
+
+  const steps = swapSteps.map(
+    (quoteStep): TradeQuoteStep => ({
+      allowanceContract: hasApprovalStep ? swapSteps[0]?.items?.[0]?.data?.to : undefined,
+      rate: quote.data.details?.rate ?? '0',
+      buyAmountBeforeFeesCryptoBaseUnit: quote.data.details?.currencyOut?.amount ?? '0',
+      buyAmountAfterFeesCryptoBaseUnit: quote.data.details?.currencyOut?.minimumAmount ?? '0',
+      sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      buyAsset,
+      sellAsset,
+      accountNumber,
+      feeData: {
+        networkFeeCryptoBaseUnit: bnOrZero(quoteStep.items?.[0]?.data?.gas)
+          .times(quoteStep.items?.[0]?.data?.maxFeePerGas)
+          .toString(),
```

Comment:
Math is incorrect, this is doing gas limit * EIP1559 max fee per gas, should be gas limit * gas price.

Reply:
Used the quote gas amount in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03)